etl/schema.py

```diff
@@ -0,0 +1,20 @@
+from pyspark.sql.types import StructType, StringType, IntegerType
+
+
+class CsvDataset:
+    schema = (
+        StructType()
+        .add("pandas_idx", StringType(), True)
+        .add("item_id", StringType(), True)
+        .add("page_id", StringType(), True)
+        .add("page_title", StringType(), True)
+        .add("top_candidates", StringType(), True)
+        .add("instance_of", StringType(), True)
+    )
+
+
+class RawDataset(CsvDataset):
+    schema = CsvDataset.schema.add("wiki_db", StringType(), True).add(
+        "snapshot", StringType(), True
+    )
+    recommendation_schema = "array<struct<image:string,note:string,rating:double>>"
```

Comment:
I'm not happy with this boilerplate to begin with, but we should (at least) rename recommendation_schema to `top_candidates_schema`  (or similar) to reflect its purpose.

Reply:
We can address this in my patch if its okay with you

Reply:
Sorry, I did miss this comment. Happy to have your refactor in your patch, or anywhere else outside of this PR.